crypto/rand/drbg_lib.c

```diff
@@ -298,10 +298,10 @@ int RAND_DRBG_instantiate(RAND_DRBG *drbg,
      * the minimum length to accomadate the length of the nonce.
      * We do this in case a nonce is require and get_nonce is NULL.
      */
-    if (drbg->min_noncelen > 0 && drbg->get_nonce == NULL) {
+    if (drbg->nonce_required && drbg->get_nonce == NULL) {
         min_entropy += drbg->strength / 2;
-        min_entropylen += drbg->min_noncelen;
-        max_entropylen += drbg->max_noncelen;
+        min_entropylen += drbg->strength / 2;
```

Comment:
`min_entropylen` is counted in bytes, and `drbg->strength`in bits, so this can't be correct. (Thanks to aa048aef0b91, this type of error is easy to detect.).

IMHO it should be `min_entropylen += min_noncelen` , so this line can be deleted because `min_entropylen = 0`